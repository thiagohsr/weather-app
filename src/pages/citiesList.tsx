import type { NextPage } from 'next'
import Link from 'next/link';
import FavouriteCitiesList from '@features/favouriteCitiesList/FavouriteCitiesList';
import {
  Container,
  Main,
  Title,
  Description,
  CodeTag,
} from '@styles/sharedStyles';

const CitiesList: NextPage = () => {
  return(
    <Container>
      <Main>
        <Title>
          Favourite Cities List
        </Title>
        <Link href={'/'}>Navigate to home</Link>

        <FavouriteCitiesList />
      </Main>
    </Container>
  )
}

export default CitiesList;