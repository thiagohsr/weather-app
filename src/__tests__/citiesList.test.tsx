import { render, screen } from "@testing-library/react";
import CitiesList from "@pages/citiesList";
import { renderWithProviders } from "@helpers/test-store-utils";
import { citiesList } from './fixtures/citiesList.fixture';
import themingWrapper from "@helpers/themingHelper";

describe("CitiesList", () => {
  it("should render CitiesList page", () => {
    const { container } = renderWithProviders(themingWrapper(<CitiesList />));

    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should renders CitiesList page heading and empty state message", async () => {
    const { container } = renderWithProviders(themingWrapper(<CitiesList />));
    const heading = screen.getByRole("heading", {
      name: /Favourite Cities List/i,
    });

    const expectedMessage = 'Do not have added cities.';

    const renderedMessage = await screen.findAllByText(expectedMessage).then(([res]) => {
      return res.innerHTML;
    })

    expect(expectedMessage).toEqual(renderedMessage);
    expect(heading).toBeInTheDocument();
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should render feature of FavouriteCitiesList", async () => {
    const { container } = renderWithProviders(themingWrapper(<CitiesList />), {
      preloadedState: {
        citiesList
      }
    });
    const listItemResult = await screen.findAllByRole("listitem");
    listItemResult.forEach((itemList, index) => {

      expect(itemList.innerHTML).toContain(Object.entries(citiesList.cities)[index][1].name);
    })
    expect(container.innerHTML).toMatchSnapshot();
  });
});
