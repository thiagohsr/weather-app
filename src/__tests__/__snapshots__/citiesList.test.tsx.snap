// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList should render CitiesList page 1`] = `"<div class="sc-f4c4221c-0 eRLUiP"><main class="sc-f4c4221c-1 gNtzww"><h1 class="sc-f4c4221c-2 mfEmD">Favourite Cities List</h1><nav class="sc-f4c4221c-12 gzyYfn"><a href="/">Search</a><a href="/citiesList">Cities list</a></nav><div style="width: 100%;" class="sc-299ea37d-0 gPQEVA"><div class="sc-299ea37d-1 efcIEP">Do not have added cities.</div></div></main></div>"`;

exports[`CitiesList should render feature of FavouriteCitiesList 1`] = `"<div class="sc-f4c4221c-0 eRLUiP"><main class="sc-f4c4221c-1 gNtzww"><h1 class="sc-f4c4221c-2 mfEmD">Favourite Cities List</h1><nav class="sc-f4c4221c-12 gzyYfn"><a href="/">Search</a><a href="/citiesList">Cities list</a></nav><div style="width: 100%;" class="sc-299ea37d-0 gPQEVA"><ul class="sc-f4c4221c-8 gWrHZL"><li class="sc-f4c4221c-10 dsfqob"><button class="sc-f4c4221c-9 fjGDyH">x</button>São José | BR<br><span class="sc-f4c4221c-11 aegYo">-1.4667, -48.6833</span></li><li class="sc-f4c4221c-10 dsfqob"><button class="sc-f4c4221c-9 fjGDyH">x</button>São José | BR<br><span class="sc-f4c4221c-11 aegYo">-28.2333, -49.1667</span></li><li class="sc-f4c4221c-10 dsfqob"><button class="sc-f4c4221c-9 fjGDyH">x</button>Città metropolitana di Roma Capitale | IT<br><span class="sc-f4c4221c-11 aegYo">41.9667, 12.6667</span></li><li class="sc-f4c4221c-10 dsfqob"><button class="sc-f4c4221c-9 fjGDyH">x</button>Recife | BR<br><span class="sc-f4c4221c-11 aegYo">-8.0539, -34.8811</span></li><li class="sc-f4c4221c-10 dsfqob"><button class="sc-f4c4221c-9 fjGDyH">x</button>São Paulo | BR<br><span class="sc-f4c4221c-11 aegYo">-23.4773, -46.6223</span></li></ul></div></main></div>"`;

exports[`CitiesList should renders CitiesList page heading and empty state message 1`] = `"<div class="sc-f4c4221c-0 eRLUiP"><main class="sc-f4c4221c-1 gNtzww"><h1 class="sc-f4c4221c-2 mfEmD">Favourite Cities List</h1><nav class="sc-f4c4221c-12 gzyYfn"><a href="/">Search</a><a href="/citiesList">Cities list</a></nav><div style="width: 100%;" class="sc-299ea37d-0 gPQEVA"><div class="sc-299ea37d-1 efcIEP">Do not have added cities.</div></div></main></div>"`;
